### DISCUSSION TITLE
How to prevent ScrollViewer from scrolling?

### DISCUSSION THREAD

**Discussion by AdamA-SVK** [AvaloniaUI/Avalonia#13541] [Q&A] [answered]
I am creating dektop app with lots of controls in one window. At the top of it is MapControl from MapsUI. I need to disable ScrollViewer scroll event while cursor is in the MapControl. In other discussion I got recommendation to use `myScrollViewer.VerticalScrollBarVisibility = Avalonia.Controls.Primitives.ScrollBarVisibility.Disabled;` to disable scrolling and `myScrollViewer.VerticalScrollBarVisibility = Avalonia.Controls.Primitives.ScrollBarVisibility.Auto;` to enable it again. The code:
AXAML
```
<ScrollViewer Name="myScrollViewer">
     <StackPanel>
          <avalonia:MapControl PointerEntered="disableScroll" PointerExited="enableScroll"/>
          <Panel Height="1500"/> <!--this is here to increase height for testing pusposes-->
     </StackPanel>
</ScrollViewer>
```
c#

```
private void disableScroll(object o, PointerEventArgs e)
{
    myScrollViewer.VerticalScrollBarVisibility = Avalonia.Controls.Primitives.ScrollBarVisibility.Disabled;
}
private void enableScroll(object o, PointerEventArgs e)
{
    myScrollViewer.VerticalScrollBarVisibility = Avalonia.Controls.Primitives.ScrollBarVisibility.Auto;
}
```

This code works well until I scroll mouse wheel in MapControl. Then scrolling in myScrollViewer won't enable anymore. 
Interesting fact about this is it only happens when myScrollViewer has space to sroll. So for example, when myScrollViewer is at the top(so there is no space to scroll up) and I zoom in the MapControl(what means scrolling up), after leaving MapControl with cursor, scrolling in myScrollViewer works. But in the same situation, when I zoom out the MapControl(I scroll down), myScrollViewer scrolling keeps disabled and can't be reenabled anymore.
Please how to fix this? Or is there any other way to prevent ScrollViewer from scrolling while cursor is scrolling over MapControl?

**Comment by timunie** [comment]
Listen for PointerWheel in a tunnel event. If sender is your MapControl, mark the event as handled for the ScrollViewer. 

*Untested*

**Reply by AdamA-SVK** [reply]
I'm quite new to Avlonia, how can I do this?

**Reply by timunie** [reply]
https://docs.avaloniaui.net/docs/next/get-started/wpf/tunnelling-events

**Comment by AdamA-SVK** [comment, marked as answer]
At the end, it was enough to just add PointerWheelChanged event to MapControl:
```
private void MapControl_PointerWheelChanged(object sender, PointerWheelEventArgs e)
{
    e.Handled = true;
}
```
